src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/TransitLayer.java

```diff
@@ -2,21 +2,23 @@
 
 import org.opentripplanner.model.Stop;
 
+import javax.annotation.Nullable;
 import java.time.LocalDate;
 import java.time.ZoneId;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class TransitLayer {
 
   /**
-   * Transit data required for routing
+   * Transit data required for routing, indexed by each local date(Graph TimeZone) it runs through.
+   * A Trip "runs through" a date if any of its arrivals or depatures is happening on that date.
    */
-  private final HashMap<LocalDate, List<TripPatternForDate>> tripPatternsForDate;
+  private final HashMap<LocalDate, List<TripPatternForDate>> tripPatternsByRunningPeriodDate;
```

Comment:
Introduction of the expression "RunningPeriodDate" may be confusing to readers. A simpler name like `TripPatternsRunningOnDate` or `TripPatternsRunningByDate` might be better, and still contrast with `TripPatternsByDate` which would index them by their "time zero reference" date.

Reply:
Also, is there any reason to use a HashMap instead of a Guava Multimap?

Reply:
I changed the name to `tripPatternsRunningOnDate`. When it comes to Multimaps I prefer using the Java `Map<,List|Set>` if I don't need the extra functionality provided by the Guava version. I have often seen the wrong guava version used (set/list), and with the common super type `Multimap` it is not always easy to see. Being consistent is also good, so it is pros and cons. This is not important to me, so I will follow whatever we decide.